sql/driver/conn.go

```diff
@@ -376,3 +522,225 @@ func (c *conn) lookupDatabase(name string) (uint32, error) {
 	}
 	return uint32(gr.ValueInt()), nil
 }
+
+func (c *conn) processColumns(desc *structured.TableDescriptor,
+	node parser.Columns) ([]structured.ColumnDescriptor, error) {
+	if node == nil {
+		return desc.Columns, nil
+	}
+
+	cols := make([]structured.ColumnDescriptor, len(node))
+	for i, n := range node {
+		switch nt := n.(type) {
+		case *parser.StarExpr:
+			return c.processColumns(desc, nil)
+		case *parser.NonStarExpr:
+			switch et := nt.Expr.(type) {
+			case *parser.ColName:
+				// TODO(pmattis): If et.Qualifier is not empty, verify it matches the
+				// table name.
+				var err error
+				cols[i], err = findColumnByName(desc, et.Name)
+				if err != nil {
+					return nil, err
+				}
+			default:
+				return nil, fmt.Errorf("unexpected node: %T", nt.Expr)
+			}
+		}
+	}
+
+	return cols, nil
+}
+
+func (c *conn) processInsertRows(node parser.InsertRows) (*rows, error) {
+	switch nt := node.(type) {
+	case parser.Values:
+		r := &rows{pos: -1}
```

Comment:
No particular drawback. This code was cribbed from the `fakedb` used by the `database/sql` tests. I'll try changing this in a subsequent PR.